core/types/rlp_backwards_compat.libevm_test.go

```diff
@@ -106,3 +109,196 @@ func testHeaderRLPBackwardsCompatibility(t *testing.T) {
 		assert.Equal(t, hdr, got)
 	})
 }
+
+func TestBodyRLPBackwardsCompatibility(t *testing.T) {
+	rng := ethtest.NewPseudoRand(0)
+
+	randTx := func() *Transaction { return NewTx(&LegacyTx{Nonce: rng.Uint64()}) }
+	randHdr := func() *Header { return &Header{ParentHash: rng.Hash()} }
+	randWithdraw := func() *Withdrawal { return &Withdrawal{Index: rng.Uint64()} }
+
+	// We build up test-case [Body] instances from the power set of each of
+	// these components.
+	txs := [][]*Transaction{
+		nil, {}, // Must be equivalent for non-optional field
+		{randTx()},
+		{randTx(), randTx()}, // Demonstrates nested lists
+	}
+	uncles := [][]*Header{
+		nil, {},
+		{randHdr()},
+		{randHdr(), randHdr()},
+	}
+	withdrawals := [][]*Withdrawal{
+		nil, {}, // Must be different for optional field
+		{randWithdraw()},
+		{randWithdraw(), randWithdraw()},
+	}
```

Comment:
nit I still don't think there is a need for a rng really, having (different) hardcoded values does the same job with a bit less code imo:

```suggestion
	// We build up test-case [Body] instances from the power set of each of
	// these components.
	txs := [][]*Transaction{
		nil, {}, // Must be equivalent for non-optional field
		{NewTx(&LegacyTx{Nonce: 1})},
		{NewTx(&LegacyTx{Nonce: 2}), NewTx(&LegacyTx{Nonce: 3})}, // Demonstrates nested lists
	}
	uncles := [][]*Header{
		nil, {},
		{&Header{ParentHash: common.Hash{1}}},
		{&Header{ParentHash: common.Hash{2}}, &Header{ParentHash: common.Hash{3}}},
	}
	withdrawals := [][]*Withdrawal{
		nil, {}, // Must be different for optional field
		{&Withdrawal{Index: 1}},
		{&Withdrawal{Index: 2}, &Withdrawal{Index: 3}},
	}
```

Reply:
Done but slightly differently so not accepting the suggestion directly.